hubble: fix Hubble Relay BASE_IMAGE

kaworu opened the pull request:
https://github.com/cilium/cilium/commit/95a4d37394be9010f9fa5691809a1ee3b8553178 ("hubble-relay: use distroless as the base image and run as non-root") attempted to use distroless as base image for Hubble Relay instead of `scratch`.

However, when running `make docker-hubble-relay-image` the image would be built with `--build-arg BASE_IMAGE=scratch` effectively overriding the base image "back" to `scratch`.

This patch make it so `BASE_IMAGE` is only overridden when set, and honor the Dockerfile's `BASE_IMAGE` otherwise.

Fix #23374, #23533

kaworu committed:
build: don't set BASE_IMAGE to scratch when it is empty

95a4d37394be9010f9fa5691809a1ee3b8553178 ("hubble-relay: use distroless
as the base image and run as non-root") attempted to use distroless as
base image for Hubble Relay instead of scratch.

However, when running `make docker-hubble-relay-image` the image would
be built with `--build-arg BASE_IMAGE=scratch` effectively overriding
the base image "back" to scratch.

This patch make it so BASE_IMAGE is only overridden when set, and honor
the Dockerfile's BASE_IMAGE otherwise.

Signed-off-by: Alexandre Perrin <alex@isovalent.com>

kaworu committed:
Revert "hubble-relay: set WORKDIR to nonroot home"

This reverts commit fc2ce0f4fc04a62ec4027d1f8986b4065601cf56.

Hubble Relay's base image, distroless nonroot, already set WORKDIR to
/home/nonroot so setting it in the Hubble Relay Dockerfile is a no-op:

    % docker image inspect -f '{{.Config.WorkingDir}}' gcr.io/distroless/static-debian11:nonroot
    /home/nonroot

Signed-off-by: Alexandre Perrin <alex@isovalent.com>

rolinh reviewed (approved):
Thanks Alex, LGTM!
Note that `BASE_IMAGE` being overwritten to `scratch` was never caught up as all images using `BASE_IMAGE` as `ARG` were actually based on `scratch` up to recently when Hubble Relay's base image was changed to distroless static.

kaworu committed:
helm: add gops configuration for Hubble Relay

Before this patch, it was not possible to disable gops for Hubble Relay
through Helm.

Signed-off-by: Alexandre Perrin <alex@isovalent.com>
